packages/connectivity/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -578,4 +583,21 @@ Possible alternatives for such technical user authentication are BasicAuthentica
       this.getSubscriberDestinationService()
     );
   }
+
+  private async addTrustStoreConfiguration(
+    destination: Destination,
+    origin: DestinationOrigin
+  ) {
+    if (destination.originalProperties?.TrustStoreLocation) {
+      const trustStoreCertificate = await fetchSubaccountCertificate(
+        this.destinationServiceCredentials.uri,
+        origin === 'subscriber'
+          ? this.subscriberToken!.serviceJwt.encoded
+          : this.providerServiceToken.encoded,
+        destination.originalProperties.TrustStoreLocation
+      );
```

Comment:
I assume this can fail, is there any error handling around it, that is not visible here?